src/declarations/stencil-private.ts

```diff
@@ -199,10 +200,16 @@ export interface RollupResults {
 export interface BuildCtx {
   buildId: number;
   buildResults: CompilerBuildResults;
+  buildStats?: CompilerBuildStats | { diagnostics: Diagnostic[] };
   buildMessages: string[];
   bundleBuildCount: number;
   collections: Collection[];
   compilerCtx: CompilerCtx;
+  esmBrowserComponentBundle: BundleModule[];
+  esmComponentBundle: BundleModule[];
+  es5ComponentBundle: BundleModule[];
+  systemComponentBundle: BundleModule[];
+  commonJsComponentBundle: BundleModule[];
```

Comment:
Consider marking these as `readonly`:
```suggestion
  esmBrowser: ReadonlyArray<CompilerBuildStatBundle>;
  esm: ReadonlyArray<CompilerBuildStatBundle>;
  es5: ReadonlyArray<CompilerBuildStatBundle>;
  system: ReadonlyArray<CompilerBuildStatBundle>;
  commonjs: ReadonlyArray<CompilerBuildStatBundle>;
```

Reply:
although that means you'd have to make some changes to `build-stats.ts`, which I added to that file as well